aspnetcore/fundamentals/minimal-apis/includes/openapi9.md

```diff
@@ -0,0 +1,126 @@
+
+* Support for generating OpenAPI documents at runtime and accessing them via an endpoint on the application
+* Support for "transformer" APIs that allow modifying the generated document
+* Support for generating OpenAPI documents at buildtime and serializing them to disk
+
+`Microsoft.AspNetCore.OpenApi` is added as a PackageReference to a project file:
+
+[!code-xml[](~/fundamentals/minimal-apis/9.0-samples/WebMinOpenApi/projectFile.xml?highlight=10)]
+
+## Describe response types
+
+OpenAPI supports providing a description of the responses returned from an API. Minimal APIs support three strategies for setting the response type of an endpoint:
+
+* Via the [`Produces`](/dotnet/api/microsoft.aspnetcore.http.openapiroutehandlerbuilderextensions.produces) extension method on the endpoint
+* Via the [`ProducesResponseType`](/dotnet/api/microsoft.aspnetcore.mvc.producesresponsetypeattribute) attribute on the route handler
+* By returning [`TypedResults`](/dotnet/api/microsoft.aspnetcore.http.typedresults) from the route handler
+
+The `Produces` extension method can be used to add `Produces` metadata to an endpoint. When no parameters are provided, the extension method populates metadata for the targeted type under a `200` status code and an `application/json` content type.
+
+```csharp
+app
+    .MapGet("/todos", async (TodoDb db) => await db.Todos.ToListAsync())
+    .Produces<IList<Todo>>();
+```
+
+Using [`TypedResults`](/dotnet/api/microsoft.aspnetcore.http.typedresults) in the implementation of an endpoint's route handler automatically includes the response type metadata for the endpoint. For example, the following code automatically annotates the endpoint with a response under the `200` status code with an `application/json` content type.
+
+```csharp
+app.MapGet("/todos", async (TodoDb db) =>
+{
+    var todos = await db.Todos.ToListAsync());
+    return TypedResults.Ok(todos);
+});
+```
+
+### Set responses for `ProblemDetails`
+
+When setting the response type for endpoints that may return a ProblemDetails response, the [`ProducesProblem`](/dotnet/api/microsoft.aspnetcore.http.openapiroutehandlerbuilderextensions.producesproblem) extension method or [`TypedResults.Problem`](/dotnet/api/microsoft.aspnetcore.http.typedresults.problem?) can be used to add the appropriate annotation to the endpoint's metadata.
+
+When there are no explicit annotations provided by one of the strategies above, the framework attempts to determine a default response type by examining the signature of the response. This default response is populated under the `200` status code in the OpenAPI definition.
+
+### Multiple response types
+
+If an endpoint can return different response types in different scenarios, you can provide metadata in the following ways:
+
+* Call the [`Produces`](/dotnet/api/microsoft.aspnetcore.http.openapiroutehandlerbuilderextensions.produces) extension method multiple times, as shown in the following example:
```

Comment:
```suggestion
* Call the [`Produces`](/dotnet/api/microsoft.aspnetcore.http.openapiroutehandlerbuilderextensions.produces) extension method multiple times, as shown in the following example:
```

Reply:
Did you mean to recommend something here? It doesn't look like there is actually a diff in the suggestion.

Reply:
Update: I'm gonna assume this was meant to be a recommendation to make it an xref like the other comments.